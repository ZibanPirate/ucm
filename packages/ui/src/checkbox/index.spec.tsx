import { fireEvent, render, screen } from "@testing-library/react";
import { FC, useState } from "react";

import { CheckBox } from ".";

describe(`Testing component '${CheckBox.name}' :`, () => {
  it(`should render it unchecked then check it`, () => {
    const FormComponent: FC = () => {
      const [checked, setChecked] = useState(false);
      return (
        <CheckBox
          data-testid="bmw-checkbox"
          label="BMW"
          checked={checked}
          onChange={(newChecked) => setChecked(newChecked)}
        />
      );
    };

    const { container } = render(<FormComponent />);
    expect(container).toMatchSnapshot("Not checked");

    const checkBox = screen.getByTestId("bmw-checkbox");
    fireEvent.click(checkBox);
    expect(container).toMatchSnapshot("Checked");
  });

  it(`should render it with no label and no initial checked state`, () => {
    const { container } = render(<CheckBox data-testid="checkbox" />);
    const checkBox = screen.getByTestId("checkbox");
    fireEvent.click(checkBox);
    expect(container).toMatchSnapshot();
  });
});
