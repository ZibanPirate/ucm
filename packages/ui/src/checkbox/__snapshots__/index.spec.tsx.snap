// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'CheckBox' : should render it unchecked then check it: Checked 1`] = `
<div>
  <div>
    <input
      data-testid="bmw-checkbox"
      id="BMW"
      name="BMW"
      type="checkbox"
    />
    <label
      for="BMW"
    >
      BMW
    </label>
  </div>
</div>
`;

exports[`Testing component 'CheckBox' : should render it unchecked then check it: Not checked 1`] = `
<div>
  <div>
    <input
      data-testid="bmw-checkbox"
      id="BMW"
      name="BMW"
      type="checkbox"
    />
    <label
      for="BMW"
    >
      BMW
    </label>
  </div>
</div>
`;

exports[`Testing component 'CheckBox' : should render it with no label and no initial checked state 1`] = `
<div>
  <div>
    <input
      data-testid="checkbox"
      id=""
      name=""
      type="checkbox"
    />
    <label
      for=""
    />
  </div>
</div>
`;
